python/cuspatial/cuspatial/io/geoseries_reader.py

```diff
@@ -0,0 +1,284 @@
+                num_rings = 1
+                rings_current = offsets["polygons"]["rings"][-1]
+                offsets["polygons"]["rings"].append(
+                    len(geometry.exterior.coords) * 2 + rings_current
+                )
+                for interior in geometry.interiors:
+                    rings_current = offsets["polygons"]["rings"][-1]
+                    offsets["polygons"]["rings"].append(
+                        len(interior.coords) * 2 + rings_current
+                    )
+                    num_rings = num_rings + 1
+                current = offsets["polygons"]["polygons"][-1]
+                offsets["polygons"]["polygons"].append(num_rings + current)
+            elif isinstance(geometry, MultiPolygon):
+                current = offsets["polygons"]["polygons"][-1]
+                offsets["polygons"]["mpolys"].append(
+                    len(offsets["polygons"]["polygons"]) - 1
+                )
+                for poly in geometry:
+                    current = offsets["polygons"]["polygons"][-1]
+                    num_rings = 1
+                    rings_current = offsets["polygons"]["rings"][-1]
+                    offsets["polygons"]["rings"].append(
+                        len(poly.exterior.coords) * 2 + rings_current
+                    )
+                    for interior in poly.interiors:
+                        rings_current = offsets["polygons"]["rings"][-1]
+                        offsets["polygons"]["rings"].append(
+                            len(interior.coords) * 2 + rings_current
+                        )
+                        num_rings = num_rings + 1
+                    offsets["polygons"]["polygons"].append(num_rings + current)
+                offsets["polygons"]["mpolys"].append(
+                    len(offsets["polygons"]["polygons"]) - 1
+                )
+        return offsets
+
+    def _read_geometries(self, geoseries, offsets):
+        """
+        Creates a set of buffers sized to fit all of the geometries and
+        iteratively populates them with geometry coordinate values.
+
+        Parameters
+        ----------
+        geoseries : A GeoPandas GeoSeries object.
+        offsets : The set of offsets that correspond to the geoseries argument.
+        """
+        buffers = {
+            "points": cudf.Series(cp.zeros(offsets["points"][-1])),
+            "multipoints": cudf.Series(cp.zeros(offsets["multipoints"][-1])),
+            "lines": cudf.Series(cp.zeros(offsets["lines"][-1])),
+            "polygons": {
+                "polygons": cudf.Series(
+                    cp.zeros(len(offsets["polygons"]["polygons"]))
+                ),
+                "rings": cudf.Series(
+                    cp.zeros(len(offsets["polygons"]["rings"]))
+                ),
+                "coords": cudf.Series(
+                    cp.zeros(offsets["polygons"]["rings"][-1])
+                ),
+            },
```

Comment:
Do you need zero-initialized memory here? This is more expensive than allocating empty buffers.

Reply:
I don't need zero-initialized memory. I don't know how to create an empty buffer of length `n` inside of a `cudf.Series`.

Reply:
`cudf.core.column.column_empty`: https://github.com/rapidsai/cudf/blob/846e6b695b2e3c9ba85303658e1f2fea08c0475d/python/cudf/cudf/core/column/column.py#L1442

Reply:
I switched it to column_empty, I don't see a significant performance change though.